src/reader/redis.clj

```diff
@@ -2,7 +2,11 @@
   (:require [taoensso.carmine :as car]
             [reader.config :as c]))
 
-(def server-conn {:pool {}, :spec {:host (c/config :redis-host)}})
+(defonce server-conn (atom nil))
+
+(defn set-conn-opts
```

Comment:
By convention, functions with dangerous side effects are suffixed with `!`. If you are keen on using this convention everywhere, then this function should be suffixed with `!`. As a rule of thumb, functions which call other functions that are suffixed with `!` should themselves be suffixed with `!`. 

Either that, or remove the `!` suffix from all of your functions for consistency. i.e. remove it from `start-app!`.

Reply:
dangerous side effects vs side effects?

Reply:
Some operations like SQL `select`s, although technically impure, don't actually mutate anything and so they aren't as dangerous as an `update`.

Reply:
so in this ns `set-to-cache`, `delete-key` and `delete-all-keys` functions warrant `!` but not `get-from-cache`